### ISSUE TITLE
Payment failure triggers channel closure

### ISSUE THREAD

**Issue by targon** [lightningnetwork/lnd#3032]
### Background

I am trying to send two payments from my lnd node to the ACINQ (03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f) node. The payments were created by lightning loop. Several routes are tried and fail.
Then lnd received this error from peer: "cannot sign when there are no changes". The other side force-closes the channel. As the channel is closed, the HTLC are still in flight and I need to wait for the timeout to recover the coins.

```
May 01 17:34:36 lnd[24172]: 2019-05-01 17:34:36.664 [ERR] CRTR: Attempt to send payment 3631efbaddf6c2824542ffd632daf0647ac0445167934e837bdae27a028875a3 failed: FeeInsufficient(htlc_amt==14791014 mSAT, update=(lnwire.ChannelUpdate) {                                                                               
May 01 17:34:36 lnd[24172]:  Signature: (lnwire.Sig) (len=64 cap=64) {                                                                                  
May 01 17:34:36 lnd[24172]:   00000000  b1 23 20 c2 5d 61 9f 37  90 72 fe 61 ba 15 44 c2  |.# .]a.7.r.a..D.|
May 01 17:34:36 lnd[24172]:   00000010  9b 0c 93 31 d6 ed 99 38  db 23 d2 bc e2 cc e2 86  |...1...8.#......|                                             
May 01 17:34:36 lnd[24172]:   00000020  61 e3 f8 e1 86 6a f1 9a  a3 6f 59 da 2f c0 7f 37  |a....j...oY./..7|                                             
May 01 17:34:36 lnd[24172]:   00000030  d4 2b ad 0c d5 41 8e 35  84 d9 59 72 9c e7 e4 53  |.+...A.5..Yr...S|                                             
May 01 17:34:36 lnd[24172]:  },                                                                                                                          
May 01 17:34:36 lnd[24172]:  ChainHash: (chainhash.Hash) (len=32 cap=32) 000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f,              
May 01 17:34:36 lnd[24172]:  ShortChannelID: (lnwire.ShortChannelID) 573938:1426:1,                                                                      May 01 17:34:36 lnd[24172]:  Timestamp: (uint32) 1556724664,                        
May 01 17:34:36 lnd[24172]:  MessageFlags: (lnwire.ChanUpdateMsgFlags) 00000001,                                                                         May 01 17:34:36 lnd[24172]:  ChannelFlags: (lnwire.ChanUpdateChanFlags) 00000001,
May 01 17:34:36 lnd[24172]:  TimeLockDelta: (uint16) 30,                                                                                                 May 01 17:34:36 lnd[24172]:  HtlcMinimumMsat: (lnwire.MilliSatoshi) 1000 mSAT,                                                                          
May 01 17:34:36 lnd[24172]:  BaseFee: (uint32) 0,                                                                                                        May 01 17:34:36 lnd[24172]:  FeeRate: (uint32) 2,                     
May 01 17:34:36 lnd[24172]:  HtlcMaximumMsat: (lnwire.MilliSatoshi) 7920000000 mSAT,                                                                     May 01 17:34:36 lnd[24172]:  ExtraOpaqueData: ([]uint8) <nil>                                               
May 01 17:34:36 lnd[24172]: }                                                                                                                           
May 01 17:34:39 lnd[24172]: 2019-05-01 17:34:39.378 [ERR] HSWC: ChannelLink(574105:489:0) Failing link: ChannelPoint(298cd458a2a5c42a6c0658c627212929566a5355db74bc6e13d993085892d7b3:0): received error from peer: cannot sign when there are no changes                                                               
May 01 17:34:39 lnd[24172]: 2019-05-01 17:34:39.378 [ERR] HSWC: ChannelLink(574105:489:0) link failed, exiting htlcManager
May 01 17:34:39 lnd[24172]: 2019-05-01 17:34:39.378 [INF] HSWC: ChannelLink(298cd458a2a5c42a6c0658c627212929566a5355db74bc6e13d993085892d7b3:0) has exited                                                                                                                                                              
May 01 17:34:39 lnd[24172]: 2019-05-01 17:34:39.378 [INF] HSWC: Removing channel link with ChannelID(b3d792580893d9136ebc74db55536a5629292127c658066c2ac4a5a258d48c29)
May 01 17:34:39 lnd[24172]: 2019-05-01 17:34:39.378 [INF] HSWC: ChannelLink(298cd458a2a5c42a6c0658c627212929566a5355db74bc6e13d993085892d7b3:0) is stopping                                                                                                                                                            
May 01 17:34:40 lnd[24172]: 2019-05-01 17:34:40.911 [INF] LNWL: Inserting unconfirmed transaction ed554fdb94c1ad12976337c445c7b54a992f3136de521db7421d7ab0a5c87b24
```

### Your environment

* version of `lnd`
lnd version 0.6.0-beta commit=v0.6.0-beta-107-g42b081bb37ce2cba3ac0452ae73407fbfde4a1d6
* which operating system (`uname -a` on *Nix)
Linux 5.0.0-13-generic #14-Ubuntu SMP Mon Apr 15 14:59:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* version of `btcd`, `bitcoind`, or other backend
Bitcoin Core Daemon version v0.17.0.1
* any other relevant environment details
These two payments were in flight:
lnbc9924270n1pwvn0rfpp550rrhuhvl6cwrkprzz4k52a6fu85jpkdq468y9u0ev27mue8zhgsdq8wdmkzuqcqzryxq97zvuqrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z96xqqqg9cqqyqqqqlgqqqqqqgq9qrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9aryqq3fqqqqqqqqlgqqqqqqgq9qrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9t3vqq8dsqqqqqqqlgqqqqqqgqjqrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9aryqq38qqqyqqqqlgqqqqqqgq9qz4julgyht7eerngjj5mzrerum60qknpf22a277k7lpgu50297v04tmrf3d4288vzx2y96jeunc7dsucm54ujqgk0w8ka0pshz24nf6cq9wf2cv
lnbc147900n1pwvn0rfpp5xcc7lwka7mpgy32zlltr9khsv3avq3z3v7f5aqmmmt385q5gwk3sdq2wpex2urp0ycqzryxq97zvuqrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z96xqqqg9cqqyqqqqlgqqqqqqgq9qrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9aryqq3fqqqqqqqqlgqqqqqqgq9qrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9t3vqq8dsqqqqqqqlgqqqqqqgqjqrzjqgwf02g2gy0l9vgdc25wxt0z72wjlfyagxlmk54ag9hyvrdsw37s6z9aryqq38qqqyqqqqlgqqqqqqgq9qfkn42jv7cnr76r8an34kllg2krld7rxgsydz4pkdrd8g4ewap3uxpqwqzlr8lug82smxtg2kka7vhyl7prud2eywd4xghxphv8vvjjsqx293a2

### Steps to reproduce

Create channel with ACINQ (0.02 BTC)
loop out 1000000

### Expected behaviour

Channel stays open even if payments fail

### Actual behaviour

Channel is force-closed by the other side

**Comment by targon** [comment]
Looks like #1127, although that issue was claimed to be fixed long ago.

**Comment by Roasbeef** [comment]
The force close is due to acinq force closing the channel. The underlying issue will be resolved by #3012.
